### PR TITLE
302 print version of package in log and api metadata

### PR DESCRIPTION
**Pull request by Acruve15**
Solving [issue 302](https://github.com/mlco2/codecarbon/issues/302)

### CODE REVIEW

File: codecarbon/__init__.py [hunk 1]
```diff
@@ -10,4 +11,4 @@
 
 __all__ = ["EmissionsTracker", "OfflineEmissionsTracker", "track_emissions"]
 
-__version__ = "2.1.3"
+__version__ = __version__
```
Comment (Fenkiou) [inline codecarbon/__init__.py:14]: You can just remove that line no?

File: examples/requirements-examples.txt [hunk 2]
```diff
@@ -2,3 +2,4 @@ torch==1.8.1
 tensorflow
 keras-tuner
 torchvision
+sklearn
```
Comment (Fenkiou) [inline examples/requirements-examples.txt:5]: How is this related to the issue you’re solving?

File: carbonserver/carbonserver/api/infra/database/sql_models.py [hunk 3]
```diff
@@ -41,6 +41,7 @@ class Run(Base):
     experiment_id = Column(UUID(as_uuid=True), ForeignKey("experiments.id"))
     os = Column(String, nullable=True)
     python_version = Column(String, nullable=True)
+    codecarbon_version = Column(String, nullable=True)
```
Comment (benoit-cty) [inline carbonserver/carbonserver/api/infra/database/sql_models.py:44]: I'm wondering if we need a more generic metadata, like:
- `client_type` to store `codecarbon package`
- `client_version` to store `2.1.3`
But maybe nobody else will use CodeCarbon API to store data.
@SaboniAmine Any idea ?

Comment (SaboniAmine) [reply carbonserver/carbonserver/api/infra/database/sql_models.py:44]: As we only have data retrieved from the package client at the moment, I guess a single column is enough.
If needed we could change this in the future.

Comment (Acruve15) [reply carbonserver/carbonserver/api/infra/database/sql_models.py:44]: Or we could have a column `metadata` and store the metadata as a dictionary?

Comment (benoit-cty) [reply carbonserver/carbonserver/api/infra/database/sql_models.py:44]: It will be difficult to query it, no ? I prefer to keep specific field, but it's a good idea if we want to be open to custom field.

File: examples/requirements-examples.txt (hunk 2 above)
Comment (benoit-cty) [reply examples/requirements-examples.txt:5]: @SaboniAmine Do you think it's a problem to add an Sklearn test ?

Sklearn is smaller that Pytorch so it may be useful.

**Review by benoit-cty — APPROVED** [review]
It's OK for me.

File: carbonserver/carbonserver/database/alembic/versions/298059b19bde_add_codecarbon_version.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+# revision identifiers, used by Alembic.
+revision = "298059b19bde"
+down_revision = "edcd10edf11d"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column("runs", sa.Column("codecarbon_version", sa.String))
```
Comment (SaboniAmine) [inline carbonserver/carbonserver/database/alembic/versions/298059b19bde_add_codecarbon_version.py:19]: We may add a default value here, such as "-1".

**Review by SaboniAmine — APPROVED** [review]
LGTM! Default value as commented is welcome but not mandatory.
Thanks @Acruve15 & Benoît for this addition!

File: carbonserver/carbonserver/database/alembic/versions/298059b19bde_add_codecarbon_version.py (hunk 4 above)
Comment (Acruve15) [reply carbonserver/carbonserver/database/alembic/versions/298059b19bde_add_codecarbon_version.py:19]: @SaboniAmine I will add the default value in the evening :)
